jsonschema/src/keywords/custom_keyword.rs

```diff
@@ -0,0 +1,71 @@
+use crate::compilation::context::CompilationContext;
+use crate::compilation::options::CustomKeywordDefinition;
+use crate::keywords::CompilationResult;
+use crate::paths::{InstancePath, JSONPointer, PathChunk};
+use crate::validator::Validate;
+use crate::ErrorIterator;
+use serde_json::{json, Value};
+use std::fmt::{Display, Formatter};
+use std::sync::Arc;
+
+// An Arc<Value> is used so the borrow checker doesn't need explicit lifetime parameters.
+// This would pollute dependents with lifetime parameters.
+pub(crate) type CustomValidateFn =
```

Comment:
I see that the `CustomIsValidFn` function uses regular references, do we need `Arc` here? I.e. how would the signature look like otherwise as you mention extra lifetimes